testonly/hammer/hammer.go

```diff
@@ -455,6 +451,9 @@ func (s *hammerState) retryOp(ctx context.Context, fn mapOperationFn, opName str
 		rspLatency.Observe(time.Since(start).Seconds(), s.label(), opName)
 	}(time.Now())
 
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Hm, by the look of it, `cancel()` in the end of this function doesn't make any difference, does it?

Reply:
I don't think it makes any difference, but wanted to keep this a pure refactoring PR.

I'll follow up with a wee tidy.